casaos-fix-docker-api-version/run.sh

```
@@ -1271,32 +1326,50 @@ downgrade_docker() {
-    echo "  Waiting for containerd... ($i/10)"
+    echo "  Waiting for containerd... ($i/15)"
     sleep 1
   done
   
   if [ "$containerd_ready" = false ]; then
     echo "⚠ WARNING: containerd may not be fully ready, proceeding anyway..."
   fi
+  
+  # Verify containerd binary version after restart
+  echo "Verifying containerd version..."
+  local containerd_version=$(timeout 5 containerd --version 2>/dev/null | head -n1 || echo "unknown")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)